lua/terrortown/menus/gamemode/equipment.lua

```diff
@@ -16,6 +16,10 @@ CLGAMEMODEMENU.isInitialized = false
 CLGAMEMODEMENU.langConVar = nil
 CLGAMEMODEMENU.lang = nil
 
+function CLGAMEMODEMENU:Initialize()
+	self:EnableSearchbar(true)
```

Comment:
Well the main idea was to let the developer have the flexibility to just use a default searchbar like here by just enabling it. But if he wants to change it, he easily can set the searchfunction manually.
Im absolutely no fan of those class overwrites you did for example in the submenu with shouldShow. Generally I like it more if you can just set a simple variable with a setter like self:shouldShow(enabled) and the backend handles it for you. Otherwise you have to write all logic in that overwrite instead of just calling a setter somewhere in your code where you need it.

But I agree, that there should be a consistent style, so I can change it into a class overwrite for 
-hasSearchBar
-getSearchBar without setter but an overwrite

Reply:
I see your point. Maybe someone else can voice his opinion here as well. I first encoutered this style in Minecraft Modding and I really liked it. This is the reason I still stick to it.

What's the difference between GetSearchBar and HasSearchBar?

Reply:
getSearchFunction i meant. See in the comment below.